site/index.html

```diff
@@ -52,50 +55,158 @@
         },
         plugins: [
           function (hook, vm) {
+            // Process Markdown directives appropriately.
             hook.beforeEach(function (content) {
-              // Match title blocks like the following:
-              // +++
-              // title: Page Title
-              // +++
-              titleBlock = /^[\+\-]{3}[\s\S]*^[\+\-]{3}$/m;
-              hugoHideDirectives = /{{% hide %}}.+?{{% \/hide %}}/gms;
-              asciinemaDirective = /{{<\s*asciinema.+key="(.+?)".+}}/g;
-              hugoDirectiveTags = /{{.*}}/g;
-              titleMatch = content.match(/title:\s*["'](.*)["']/);
-
-              titleHeading = titleMatch ? "# " + titleMatch[1] : "";
-
-              content = content
-                .replace(titleBlock, titleHeading)
-                .replace(asciinemaDirective, (_, fileName) => `<asciinema-player src="${ window.location.origin}/static/casts/${fileName}.cast"></asciinema-player>`)
-                .replace(hugoHideDirectives, "")
-                .replace(hugoDirectiveTags, "");
+              content = processAsciinemaTags(content);
+
+              // Until all source markdown files stop using Hugo directives,
+              // convert here for compatibility.
+              content = convertFromHugo(content);
               return content;
             });
 
-            hook.doneEach(function() { 
-              const tocLists = Array.from(document.getElementsByClassName("sidebar-nav").item(0)?.getElementsByTagName("ul"))
-              
-              // Hide a child list if neither its parent nor any of its descendants are active.
-              tocLists.forEach(
-                  ul => (ul.parentElement.classList.contains("active") || ul.getElementsByClassName("active").length) ?
-                   ul.classList.remove("inactive") : 
-                   ul.classList.add("inactive")
-                  );
+            // Add title to book Markdown pages based on directory structure.
+            hook.beforeEach(processBookPageTitle);
+
+            // Show navigation footer for book pages.
+            hook.doneEach(showBookPageFooters);
+
+            // Process elements in the navigation sidebar.
+            hook.doneEach(function () {
+              const sidebar = document
+                .getElementsByClassName("sidebar-nav")
+                .item(0);
+
+              // Post-process UI elements like chapter number for the book outline in the sidebar.
+              processBookOutline(sidebar);
+
+              // Only show child pages for currently active page to avoid sidebar cluttering.
+              addSidebarCollapsibility(sidebar);
             });
           },
         ],
       };
+
+      function convertFromHugo(content) {

```

Comment:
why's there remnants of Hugo?

Reply:
The Hugo directives in the source markdown of each page were left until that the site content itself was revisited as part of v1. This code just handles interpreting/obscuring those directives until then and should disappear once content for the site is refreshed.